makefile.shade

```diff
@@ -780,4 +841,163 @@ functions
+                            {
+                                GetDependencies(projectJson, info.DependencyNames);
+                            }
+                        }
+                    }
+                }
+
+                info.DependencyNames.ExceptWith(info.RepositoryNames);
+                repositoryLookup.Add(info);
+            }
+
+            foreach (var info in repositoryLookup)
+            {
+                foreach (var item in info.DependencyNames)
+                {
+                    var dependency = repositoryLookup.Find(r => r.RepositoryNames.Contains(item));
+                    if (dependency != null)
+                    {
+                        // Testing has odd cyclic dependencies
+                        if (string.Equals(info.Name, "Testing", StringComparison.OrdinalIgnoreCase) &&
+                            string.Equals(dependency.Name, "aspnet.xunit", StringComparison.OrdinalIgnoreCase))
+                        {
+                            continue;
+                        }
+
+                        if (string.Equals(dependency.Name, "Testing", StringComparison.OrdinalIgnoreCase) &&
+                            (string.Equals(info.Name, "Common", StringComparison.OrdinalIgnoreCase) ||
+                            string.Equals(info.Name, "PlatformAbstractions", StringComparison.OrdinalIgnoreCase) ||
+                            string.Equals(info.Name, "Logging", StringComparison.OrdinalIgnoreCase)))
+                        {
+                            continue;
+                        }
+
+                        info.Dependencies.Add(dependency);
+                    }
+                }
+            }
+
+            return repositoryLookup.GroupBy(r => r.Order, r => r.Name).OrderBy(r => r.Key);
+        }
+
+        static void GetDependencies(string projectJsonPath, HashSet<string> dependencies)
+        {
+            var project = (JsonObject)Json.Deserialize(File.ReadAllText(projectJsonPath));
+            var dependenciesNode = project.ValueAsJsonObject("dependencies");
+            AddKeys(dependencies, dependenciesNode);
+
+            var frameworkNodes = project.ValueAsJsonObject("frameworks");
+            if (frameworkNodes != null)
+            {
+                foreach (var framework in frameworkNodes.Keys)
+                {
+                    dependenciesNode = frameworkNodes.ValueAsJsonObject(framework).ValueAsJsonObject("dependencies");
+                    AddKeys(dependencies, dependenciesNode);
+                }
+            }
+
+            AddKeys(dependencies, project.ValueAsJsonObject("commands"));
+        }
+
+        static void AddKeys(HashSet<string> target, JsonObject source)
+        {
+            if (source != null)
+            {
+                foreach (var key in source.Keys)
+                {
+                    target.Add(key);
+                }
+            }
+        }
+
+        string CreateBuildWithFlowId(string repo)
```

Comment:
`exec` shade from Sake has code to redirect standard out \ standard err to Log. I'd rather not duplicate it here.